lib/rules/no-native-reassign.js

```diff
@@ -17,6 +18,8 @@ module.exports = {
             recommended: true
         },
 
+        deprecated: true,
```

Comment:
Should this be in `docs` too? Maybe with a `replacedBy: "no-global-assign"` field?